### PR TITLE
fix: do not poll IF-MIB by default

### PR DESCRIPTION
**Pull request by omrozowicz-splunk**
# Description

This PR limits the default small walk to not include `IF-MIB` family as this is not what we always want to poll from the devices.
Additionally it adds a way to stop polling base profiles as not everyone need them.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Refactor/improvement

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist

- [x] My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] I have run pre-commit on all files before creating the PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: docs/configuration/configuring-profiles.md [hunk 1]
```diff
@@ -160,7 +160,7 @@ poller:
 ```
 
 NOTE: When small walk is configured, you can set up polling only of OIDs belonging to the walk profile varBinds. 
-Additionally, there are two MIB families that are enabled by default (we need them to create the state of the device in the database and poll base profiles): `IF-MIB` and `SNMPv2-MIB`.
+Additionally, `SNMPv2-MIB` that is enabled by default (we need them it create the state of the device in the database).
```
Comment (wojtekzyla) [inline docs/configuration/configuring-profiles.md:163]: "we need it to create"

File: docs/configuration/configuring-profiles.md [hunk 2]
```diff
@@ -160,7 +160,7 @@ poller:
...
 For example, if you've decided to use `small_walk` from the example above, you'll be able to poll only `UDP-MIB`, `IF-MIB`, and `SNMPv2-MIB` OIDs.
```
Comment (wojtekzyla) [inline docs/configuration/configuring-profiles.md:164]: shouldn't it be "poll only `UDP-MIB` and `SNMPv2-MIB` OIDs."

File: docs/configuration/configuring-profiles.md (hunk 1 above)
Comment (omrozowicz-splunk) [reply docs/configuration/configuring-profiles.md]: fixed

File: docs/configuration/configuring-profiles.md (hunk 2 above)
Comment (omrozowicz-splunk) [reply docs/configuration/configuring-profiles.md]: right, changed

**Pull request merged**